### PR TITLE
RPE operation and fusing transformation

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
 - *Initial version of **_internal_** Rotary Positional Embedding operation and fusing transformation*

### Tickets:
 - *CVS-123771*

### CODE REVIEW

**Comment by jane-intel** [comment]
TODO: consider other patterns

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            return false;  // flawed match
+        if (actual_source != potential_source && actual_source == cos_output)
+            cos_output = potential_source;
+
+        auto input = value_map.at(source);
+        auto concat_node = ov::as_type_ptr<Concat>(value_map.at(concat).get_node_shared_ptr());
+        if (!concat_node)
+            return false;
+        OPENVINO_SUPPRESS_DEPRECATED_START
+        auto split_axis_node = ov::get_constant_from_source(value_map.at(axis));
```
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:72]:
```suggestion
        auto split_axis_node = ov::util::get_constant_from_source(value_map.at(axis));
```
Is in `dev_api` validation util. Migrated anad `OPENVINO_SUPPRESS_DEPRECATED_START` macros can be removed.

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        auto value = split_axis_node->cast_vector<int64_t>();
+        if (value.size() != 1)
+            return false;
+        auto concat_axis = concat_node->get_concatenation_axis();
+        auto split_axis = value[0];
+        if (concat_axis != split_axis) {
+            if (input.get_partial_shape().is_static()) {
+                auto rank = input.get_partial_shape().rank().get_length();
+                concat_axis = concat_axis < 0 ? concat_axis + rank : concat_axis;
+                split_axis = split_axis < 0 ? split_axis + rank : split_axis;
```
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:85]: Just optional if `#include "validaition_util.hpp"` is available from `dev_api` then:
```suggestion
                concat_axis = ov::util:::normalize(concat_axis, rank);
                split_axis = ov::util:::normalize(split_axis, rank);
```

**Review by praasz — APPROVED** [review]

File: src/common/transformations/src/ov_ops/rotary_positional_embeddings.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+bool RPE::visit_attributes(ov::AttributeVisitor& visitor) {
+    INTERNAL_OP_SCOPE(internal_RoPE_visit_attributes);
+    visitor.on_attribute("axis", m_axis);
+    return true;
+}
+
+std::shared_ptr<ov::Node> RPE::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    INTERNAL_OP_SCOPE(internal_RoPE_clone_with_new_inputs);
+    return std::make_shared<RPE>(new_args.at(0), new_args.at(1), new_args.at(2), m_axis);
```
Comment (itikhono) [inline src/common/transformations/src/ov_ops/rotary_positional_embeddings.cpp:40]: could you "check_new_args_count" or similar method to check that we have the expected number of new_args (3 in this case)?

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        if (concat_axis != split_axis) {
+            if (input.get_partial_shape().is_static()) {
+                auto rank = input.get_partial_shape().rank().get_length();
+                concat_axis = ov::util:::normalize(concat_axis, rank);
+                split_axis = ov::util:::normalize(split_axis, rank);
+            }
+            if (concat_axis != split_axis)
+                return false;
+        }
+        auto rpe =
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:90]: could you copy rt info from the fused ops to the new RPE op?

File: src/common/transformations/tests/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        auto cos = make_shared<v0::Parameter>(element::f32, PartialShape::dynamic());
+        name_node_and_output(cos, "cos");
+        auto rpe = make_shared<ov::op::internal::RPE>(data, sin, cos, -1);
+        name_node_and_output(rpe, "rpe");
+        model_ref = std::make_shared<Model>(NodeVector{rpe}, ParameterVector{data, sin, cos});
+    }
+    comparator.enable(FunctionsComparator::CmpValues::NAMES);
+}
+
+TEST_F(TransformationTestsF, FuseRPESorcesAreMultiOutputed) {
```
Comment (itikhono) [inline src/common/transformations/tests/common_optimizations/fuse_rotary_positional_embeddings.cpp:64]: could you add a comment why the transformation shouldn't be applied in this case
It's hard to understand it from the test name

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/include/ov_ops/rotary_positional_embeddings.hpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace ov {
+namespace op {
+namespace internal {
+
+///
+/// \brief Rotary Positional Embeddings operation
+/// Internal operation which may change in the future
+/// \ingroup ov_ops_cpp_api
+class TRANSFORMATIONS_API RPE : public ov::op::Op {
```
Comment (ilya-lavrenov) [inline src/common/transformations/include/ov_ops/rotary_positional_embeddings.hpp:18]: RoPE is more common abbreviation for this operation.

Comment (jane-intel) [reply src/common/transformations/include/ov_ops/rotary_positional_embeddings.hpp:18]: Discussed the question offline.

**Pull request merged**